src/test/java/org/apache/accumulo/access/AccessExpressionTest.java

```diff
@@ -189,4 +198,30 @@ public void testEqualsHashcode() {
     assertEquals(ae1.hashCode(), ae3.hashCode());
     assertNotEquals(ae1.hashCode(), ae2.hashCode());
   }
+
+  @Test
+  public void testSpecificationDocumentation() throws IOException, URISyntaxException {
+    // verify AccessExpression.abnf matches what is documented in SPECIFICATION.md
+
+    // read the abnf spec, ignoring the header
+    List<String> specLinesFromAbnfFile;
+    try (
+        var abnfFileStream =
+            AccessExpression.class.getResourceAsStream("specification/AccessExpression.abnf");
+        var inputStreamReader = new InputStreamReader(abnfFileStream, UTF_8);
+        var bufferedReader = new BufferedReader(inputStreamReader)) {
+
+      Predicate<String> abnfComment = line -> line.startsWith(";");
+      Predicate<String> beforeFirstLine = abnfComment.or(String::isBlank);
+      specLinesFromAbnfFile = bufferedReader.lines().dropWhile(beforeFirstLine).collect(toList());
+    }
+
+    // grab from the markdown, but make sure to skip the markdown triple ticks
```

Comment:
IIRC, the `AccessExpression.abnf` file exists solely so that we can test that it's valid ABNF. We were unable to figure out how to embed `AccessExpression.abnf` into `SPECIFICATION.md`. Given that you were able to just pull the ABNF out of `SPECIFICATION.md`, then I think we could probably get rid of `AccessExpression.abnf` and `SpecificationTest` (or just move this test method to `SpecificationTest`.

This is a great change.

Reply:
I thought it would be useful to leave it in its own file in src/main/resources, so it could be grabbed from the jar by other projects, if they wanted to use it. As the canonical definition, I think it's useful in the jar so it can be used as a dependency.